### DISCUSSION TITLE
Working with object fields

### DISCUSSION THREAD

**Discussion by seMikel** [react-hook-form/react-hook-form#9042] [Q&A]
Hello.

I have a custom input that can be interacted with in different ways to ultimately produce a simple object with data. As a result the `value` and `onChange` props are an object and an object callback function. The issue is that calling `setValue` doesn't trigger a rerender of the input, presumably because of the note from the docs.

> Avoid including custom object into the defaultValues. eg: moment, luxon as those will lead to unexpected result during internal object clone process.

I would like to know if there is a known workaround for this, as I wasn't able to find a question for this exact scenario.

Thank you.

**Comment by bluebill1049** [comment]
you can build a controlled component to host the object data and then wrapped with controller/useController to simply collect filed value as object.

**Reply by seMikel** [reply]
Sorry for the delay, I believe I am already doing what you are describing. I have a Controller with the custom input in question being returned in the Controller's `render` prop. The issue is that even though inputting by hand works seemlessly, calling setValue does not trigger the function in the `render` prop.
